src/templates/script/parseScriptTemplates.ts

```diff
@@ -142,6 +143,43 @@ function parseScriptSetting(data: object, resources: IResources, variables: IVar
     };
 }
 
+// the bindings are similar to v1, but the input is coming from the trigger template so we need to account for that as well
+export function parseScriptSettingV2(data: object | undefined, resources: IResources, input: IInput): IBindingSetting {
+    const rawSetting: IRawSetting = <IRawSetting>data;
+    const enums: IEnumValue[] = [];
+    if (rawSetting && rawSetting.enum) {
+        for (const ev of rawSetting.enum) {
+            enums.push({
+                value: getResourceValue(resources, ev.value),
+                displayName: getResourceValue(resources, ev.display)
+            });
+        }
+    }
+
+    return {
+        name: rawSetting ? getResourceValue(resources, rawSetting.name) : input.paramId,
+        resourceType: rawSetting ? rawSetting.resource : undefined,
```

Comment:
Why not this for these and below?
```suggestion
        resourceType: rawSetting?.resource,
```